### PR TITLE
feat: add support for object deconstructing.

### PR DESCRIPTION
**Pull request by vjerci**
## Description
This pr adds support for object deconstructing.
It adds minimal support in a way we only support first level of object deconstructing.

---

### Supported findings

-  `const` will detect both `let` and `const`, they are considered the same
- `var` needs to be specified - it's different than `const` or `let` 

Rule:
```
const { $<!>$<_> } = req.params
```
Matching testcases:
```
const { user } = req.params
let { user } = req.params
```

Rule:
```
var { $<!>$<_> } = req.params
```
Matching testcases:
```
var { user } = req.params
```

---

### Unsuported findings
Nested cases
- we don't support cases like these
```
var { user: { admin, superuser } } = req.params
```

---

## Related
https://github.com/Bearer/bearer/issues/698

## Checklist

- [x] I've added test coverage that shows my fix or feature works as expected.
- [x] I've updated or added documentation if required.
- [x] I've included usage information in the description if CLI behavior was updated or added.
- [x] PR title follows [Conventional Commits](https://www.conventionalcommits.org/) format

### CODE REVIEW

File: new/detector/composition/javascript/testdata/testcases/object-deconstructing/deconstructing.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+const { user } = req.params;
```
Comment (cfabianski) [inline new/detector/composition/javascript/testdata/testcases/object-deconstructing/deconstructing.js:1]: Do we support things like `const { firstName,  lastName } = req.params` as well?
It's not clear to be based on what I can see

Comment (vjerci) [reply new/detector/composition/javascript/testdata/testcases/object-deconstructing/deconstructing.js:1]: we do, I've added test case for it

Comment (cfabianski) [reply new/detector/composition/javascript/testdata/testcases/object-deconstructing/deconstructing.js:1]: I don't see the test case :/
Could you send me the link to it please?

**Review by elsapet — APPROVED** [review]

**Review by cfabianski — APPROVED** [review]

**Review by cfabianski — APPROVED** [review]
LGTM. Good job

**Pull request merged**